### DISCUSSION TITLE
how to replicate results from sklearn LinearRegression

### DISCUSSION THREAD

**Discussion by samchaaa** [online-ml/river#1154] [Q&A] [answered]
Hello, I am trying to replicate the results from repeatedly calling sklearn LinearRegression at each new datapoint.

Currently I am just using this configuration: 

```
model = compose.Pipeline(
    preprocessing.StandardScaler(),
    linear_model.LinearRegression()
)
```

My results are fairly different, I am curious about the intercept_lr (this seems to be a param that sklearn doesn't have). 

How do I replicate results exactly? 

Thank you

**Comment by MaxHalford** [comment, marked as answer]
Hey there. I don't believe there is an obvious to reproduce the outputs of scikit-learn's linear regression.

However, you should be able to reproduce the results from their SGDRegressor. In fact, we even have a unit test for that.

The mathematics of batch and online linear regression are different. The online version should converge to the batch version, but there are no systematic guarantees of that.

**Reply by samchaaa** [reply]
Thanks for the prompt response.

On a different note, congrats on a really well done project. Everything is really clean.

I'm interested in contributing, I noticed the link in the readme is closed though (simpler tasks in Notion: https://www.notion.so/maxhalford/d1e86fcdf21e4deda16eedab2b3361fb?v=e74f58d5a1104e0ab1b29ba0761a7610)

**Reply by MaxHalford** [reply]
> On a different note, congrats on a really well done project. Everything is really clean.

Thanks for saying that! It means a lot.

> I'm interested in contributing, I noticed the link in the readme is closed though (simpler tasks in Notion: [notion.so/maxhalford/d1e86fcdf21e4deda16eedab2b3361fb?v=e74f58d5a1104e0ab1b29ba0761a7610](https://www.notion.so/maxhalford/d1e86fcdf21e4deda16eedab2b3361fb?v=e74f58d5a1104e0ab1b29ba0761a7610))

Good catch. We actually [switched](https://github.com/orgs/online-ml/projects/3?query=is%3Aopen+sort%3Aupdated-desc) to GitHub Projects. I've removed the "is simple" tag for now. But feel free to work on anything you want :)
